core/chain-configs/src/client_config.rs

```diff
@@ -503,16 +567,7 @@ pub struct ClientConfig {
     pub block_header_fetch_horizon: BlockHeightDelta,
     /// Garbage collection configuration.
     pub gc: GCConfig,
-    /// Accounts that this client tracks.
-    pub tracked_accounts: Vec<AccountId>,
-    /// Track shards that should be tracked by given validator.
-    pub tracked_shadow_validator: Option<AccountId>,
-    /// Shards that this client tracks.
-    pub tracked_shards: Vec<ShardId>,
-    /// Rotate between these sets of tracked shards.
-    /// Used to simulate the behavior of chunk only producers without staking tokens.
-    /// This field is only used if `tracked_shards` is empty.
-    pub tracked_shard_schedule: Vec<Vec<ShardId>>,
+    pub tracked_config: TrackedShardsConfig,
```

Comment:
In terms of feature name, I was thinking we should name this either `tracked_shards` or `tracked_shards_config`. The `_config` might be redundant as we already know it's part of config, but should be fine to have it there.

The current `tracked_config` is super unintuitive and it initially didn't make sense to me what it exactly was until I read `TrackedShardsConfig`.